mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -1145,6 +1145,46 @@ def Rock_AccelGemmOp:
   let hasVerifier = 1;
 }
 
+// blockwise_broadcasting_reduction
+def Rock_BlockwiseBroadcastReduceOp:
+    Rock_Op<"blockwise_broadcast_reduce">,
+    Arguments<(ins MemRefOf<GemmOutputTypes>:$input,
+                   MemRefOf<LdsBufferTypes>:$workspace_buffer,
+                   MemRefOf<GemmOutputTypes>:$output,
+                   IndexAttr:$axis,
+                   ReduceMethodAttr:$reduceMethod,
+                   TransformMapArrayAttr:$inputRegView,
+                   I32Attr:$blockSize
+                   )> {
+  let summary = "Blockwise reduce operation with output broadcasting";
+  let description = [{
+    The `rock.blockwise_broadcast_reduce` op does reduce at a blockwise level and
+    to fill all elements of the input tensor with reduced values. Note that both input/output
+    tensors live in registers.
+
+    $input tensor resides in registers in threads.
+
+    $inputRegView should be register {tid, iter}
+    to virtual tensor {d0, ... , Dr , ... , dn} which is a slice of
+    a larger tensor {D0, ... , Dr , ... , Dn}. Note that the reduction
+    axis (Dr) is not sliced because it is not supported.
+
+    $output tensor resides in registers in threads. It would have
```

Comment:
I don't understand this paragraph at all, please reword.

Reply:
reworded and removed additional/unncessary info on "how to retrieve non-broadcasted" tensor which has nothing to do with this op.